packages/esm-appointments-app/src/helpers/excel.ts

```diff
@@ -4,20 +4,31 @@ import { type Appointment } from '../types';
 import { type ConfigObject } from '../config-schema';
 import { moduleName } from '../constants';
 
+type RowData = {
+  id: string; // Corresponds to the UUID of an appointment
+  identifier?: string; // Optional identifier property
+  [key: string]: any; // Allow for other dynamic properties
+};
+
 /**
  * Exports the provided appointments as an Excel spreadsheet.
  * @param {Array<Appointment>} appointments - The list of appointments to export.
+ * @param {Array} rowData - The current rows of the table as rendered in the UI.
  * @param {string} [fileName] - The name of the downloaded file
  */
 export async function exportAppointmentsToSpreadsheet(
   appointments: Array<Appointment>,
-  fileName: string = 'Appointments',
+  rowData: Array<RowData>,
+  fileName = 'Appointments',
 ) {
   const config = await getConfig<ConfigObject>(moduleName);
   const includePhoneNumbers = config.includePhoneNumberInExcelSpreadsheet ?? false;
 
   const appointmentsJSON = await Promise.all(
     appointments.map(async (appointment: Appointment) => {
+      const tableRow = rowData.find((row) => row.id === appointment.uuid);
+      const identifier = tableRow?.identifier ?? appointment.patient.identifier;
```

Comment:
```suggestion
      const matchingAppointment = rowData.find((row) => row.id === appointment.uuid);
      const identifier = matchingAppointment?.identifier ?? appointment.patient.identifier;
```

nit: maybe `matchingAppointment` conveys the intention of the code better.

Reply:
Hello @denniskigen, I have implemented the changes.